packages/svelte/src/internal/client/dom/blocks/boundary.js

```diff
@@ -21,116 +20,170 @@ import {
 import { queue_micro_task } from '../task.js';
 
 /**
- * @param {Effect} boundary
- * @param {() => void} fn
+ * @typedef {{
+ * 	 onerror?: (error: unknown, reset: () => void) => void;
+ *   failed?: (anchor: Node, error: () => unknown, reset: () => () => void) => void;
+ * }} BoundaryProps
  */
-function with_boundary(boundary, fn) {
-	var previous_effect = active_effect;
-	var previous_reaction = active_reaction;
-	var previous_ctx = component_context;
-
-	set_active_effect(boundary);
-	set_active_reaction(boundary);
-	set_component_context(boundary.ctx);
-
-	try {
-		fn();
-	} finally {
-		set_active_effect(previous_effect);
-		set_active_reaction(previous_reaction);
-		set_component_context(previous_ctx);
-	}
-}
+
+var flags = EFFECT_TRANSPARENT | EFFECT_PRESERVED | BOUNDARY_EFFECT;
 
 /**
  * @param {TemplateNode} node
- * @param {{
- * 	 onerror?: (error: unknown, reset: () => void) => void,
- *   failed?: (anchor: Node, error: () => unknown, reset: () => () => void) => void
- * }} props
- * @param {((anchor: Node) => void)} boundary_fn
+ * @param {BoundaryProps} props
+ * @param {((anchor: Node) => void)} children
  * @returns {void}
  */
-export function boundary(node, props, boundary_fn) {
-	var anchor = node;
+export function boundary(node, props, children) {
```

Comment:
if this is intended to match the signature of the `Boundary` constructor you can just type it `/** @type {typeof Boundary} */` to avoid having all of the duplicate JSDoc

Reply:
not working for me?

<img width="823" alt="image" src="https://github.com/user-attachments/assets/9c2e02a3-bd1d-49ea-9185-9521096eced1" />